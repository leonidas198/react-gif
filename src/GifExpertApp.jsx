import { useState } from "react";
import { AddCategory, GifGrid} from "./components";


export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Dragon Ball Z']);

  const onAddCategory = (newCategory) => {

    const textMayus = newCategory.toUpperCase();

    if (categories.includes( textMayus )) return; //Esto hace que no se puedan repetir nombres

    //console.log( newCategory )
    setCategories([textMayus, ...categories]); // Esto hace que agreguemos nuevos nombres
    //setCategories( c => [...c, 'Valorant'] );
  };

  return (
    <>
      <div className="container">
        
        <a href="https://amzn.to/4g4nU8F"><img className="amazon" src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRckZToY8MPfbR4AV9nMBn9boLVvdE5F0_nOg&s" alt="foto" /></a>
      </div>
      <hr />
      <h1> Gifs de Anime </h1>

      <AddCategory
        //setCategories={ setCategories }
        onNewCategory={(event) => onAddCategory(event)} //esto hace que se ejecute la funcion
      />

      

      {
        categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))

        //return <li key={ category }>{ category }</li>//este codigo es para agregar listas sin usar etiquetas li
      }
    </>
  );
};
