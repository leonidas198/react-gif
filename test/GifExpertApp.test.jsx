import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"


describe('Pruebas en el GifExpertApp', () => {

    test('debe hacer el match con el snapshot', () => {

        const { container } = render( <GifExpertApp /> );
        expect( container ).toMatchSnapshot();
        
        screen.debug();
      
    })
    
  
})
