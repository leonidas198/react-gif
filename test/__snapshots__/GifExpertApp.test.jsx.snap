// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifExpertApp debe hacer el match con el snapshot 1`] = `
<div>
  <h1>
     Gifs de Anime 
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
     
    Dragon Ball Z
     
  </h3>
  <h4>
     Cargando... 
  </h4>
  <div
    class="card-grid"
  />
</div>
`;
